### ISSUE TITLE
(READ THIS FIRST) - Correct Way to get Google Search Results

### ISSUE THREAD

**Issue by knoxcard** [jprichardson/node-google#63]
The proper way to get Google results is to use an API key.  With this module, you'll quickly run into the Google captcha issue (Google will detect an unusual amount of traffic from an IP), without using an API key.

You want to query Google directly using an API key...
`https://www.googleapis.com/customsearch/v1?key=<YOUR_GOOGLE_SEARCH_API_KEY>&q=Node JS`

Get your Google Search API Key here: https://developers.google.com/custom-search/json-api/v1/overview

This is wayyyy faster too, you get raw JSON results directly from their server in one HTTP call.  No time wasted on parsing HTML and if Google changes their HTML structure you are not dead in the water.  Also, one less NPM repo in your codebase, better security.

**Comment by Polunom** [comment]
Pretty sure you can't search the entire web with this method, as calling the custom search API requires a `cx` field. Under the (create new custom search)[https://cse.google.com/cse/create/new] page, the option to search the entire web was removed

**Comment by BeauBouchard** [comment]
Thank you :+1:

**Comment by tpickett** [comment]
i don't agree. Using the API provides a bias about your API key in the results that you receive. Yes it is faster, but is isn't the same results... 
There are workarounds for the IP request limits that will ban you. 
I don't think that one way is "correct" over another. Google wants us to use the API? then they shouldn't skew the results from said API. This is just spreading false information...

**Comment by TowhidKashem** [comment]
Thanks a lot, worked like a charm! Apparently you get 100 free searches a day for free when signing up directly with google as well.
